core/transfer/src/main/java/org/eclipse/dataspaceconnector/transfer/core/transfer/TransferProcessManagerImpl.java

```diff
@@ -219,30 +217,41 @@ private int checkProvisioned() {
     private int checkCompleted() {
 
         //deal with all the consumer processes
-        List<TransferProcess> processesInProgress = transferProcessStore.nextForState(TransferProcessStates.IN_PROGRESS.code(), batchSize);
+        var processesInProgress = transferProcessStore.nextForState(TransferProcessStates.IN_PROGRESS.code(), batchSize);
 
         for (var process : processesInProgress.stream().filter(p -> p.getType() == CONSUMER).collect(Collectors.toList())) {
-
             if (process.getDataRequest().isManagedResources()) {
-                List<ProvisionedResource> resources = process.getProvisionedResourceSet().getResources().stream().filter(this::hasChecker).collect(Collectors.toList());
-
-                // update the process once ALL resources are completed
-                if (resources.stream().allMatch(this::isComplete)) {
-                    process.transitionCompleted();
-                    monitor.debug("Process " + process.getId() + " is now " + TransferProcessStates.COMPLETED);
-                    invokeForEach(listener -> listener.completed(process));
+                var resources = process.getProvisionedResourceSet().getResources();
+                var checker = statusCheckerRegistry.resolve(process.getDataRequest().getDestinationType());
+                if (checker == null) {
+                    monitor.info(format("No checker found for process %s. The process will not advance to the COMPLETED state.", process.getId()));
+                } else if (checker.isComplete(process.getId(), resources)) {
+                    // checker passed, transition the process to the COMPLETED state
+                    transitionToCompleted(process);
                 }
             } else {
-                //no managed resources, just update
-                process.transitionCompleted();
-                monitor.debug("Process " + process.getId() + " is now " + TransferProcessStates.COMPLETED);
-                invokeForEach(listener -> listener.completed(process));
+                var checker = statusCheckerRegistry.resolve(process.getDataRequest().getDestinationType());
+                if (checker != null) {
+                    if (checker.isComplete(process.getId(), emptyList())) {
+                        //checker passed, transition the process to the COMPLETED state automatically
+                        transitionToCompleted(process);
+                    }
```

Comment:
This will always throw an `EdcException` for unmanaged S3 buckets, i.e. when the `S3StatusChecker` is used: 
```java
        // in S3StatusChecker.java: 
        for (var resource : resources) {
              // do completion checking
        }
        throw new EdcException(format("No bucket resource was associated with...."));
```
so the exceptio is raised when an empyt list is passed.

One option would be to obtain the transfer process by its `id` and use 
```java
TransferProcess.getDataRequest().getDataDestination()
``` 
as a fallback.

The same is true for the `ObjectContainerStatusChecker`

Reply:
OK thanks, Paul and I discussed and he supplied a patch that I just committed